dpsim-models/src/EMT/EMT_Ph1_Resistor.cpp

```diff
@@ -113,3 +113,15 @@ void EMT::Ph1::Resistor::mnaCompUpdateVoltage(const Matrix &leftVector) {
 void EMT::Ph1::Resistor::mnaCompUpdateCurrent(const Matrix &leftVector) {
   (**mIntfCurrent)(0, 0) = (**mIntfVoltage)(0, 0) / **mResistance;
 }
+
+void EMT::Ph1::Resistor::stampBranchNodeIncidenceMatrix(
+    Matrix &branchNodeIncidenceMatrix) {
+  if (terminalNotGrounded(0)) {
+    branchNodeIncidenceMatrix(mBranchIdx, matrixNodeIndex(0)) = 1.0;
+  }
+  if (terminalNotGrounded(1)) {
+    branchNodeIncidenceMatrix(mBranchIdx, matrixNodeIndex(1)) = -1.0;
+  }
+}
+
+void EMT::Ph1::Resistor::setBranchIdx(UInt i) { mBranchIdx = i; }
```

Comment:
Thank you for bringing attention to this!

Traditionally, interfaces should not provide function implementations. For example, see [Wikipedia - Interface (object-oriented programming)](https://en.wikipedia.org/wiki/Interface_(object-oriented_programming))
> In [object-oriented programming](https://en.wikipedia.org/wiki/Object-oriented_programming), an interface or protocol type[[a]](https://en.wikipedia.org/wiki/Interface_(object-oriented_programming)#cite_note-1) is a [data type](https://en.wikipedia.org/wiki/Data_type) that acts as an [abstraction](https://en.wikipedia.org/wiki/Abstraction) of a [class](https://en.wikipedia.org/wiki/Class_(computer_science)). It describes a set of [method signatures](https://en.wikipedia.org/wiki/Method_signature), the implementations of which may be provided by multiple [classes](https://en.wikipedia.org/wiki/Class_(computer_programming)) that are otherwise not necessarily related to each other.

However, it seems that the concept of interfaces is evolving and having default implementations of functions in an interface is okay. For example, [in C#](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/interfaces)

> Beginning with C# 8.0, an interface may define default implementations for some or all of its members. A class or struct that implements the interface doesn't have to implement members that have default implementations. For more information, see [default interface methods](https://learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interface-implementation/default-interface-methods-versions).

Personally, I would leave it like that for now and think of an alternative solution, while further working on eigenvalue extraction feature.

Reply:
Thank you, done ✔: [Rephrase/clean up TODOs related to eigenvalue extraction ](https://github.com/sogno-platform/dpsim/commit/8ac5cc8f283dc3df2ed64df8cc99bb41a0c1dc9d)